### PR TITLE
Add organization configuration management API

### PR DESCRIPTION
**Pull request by dewniMW**
Issue: https://github.com/wso2/product-is/issues/16728

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.common/src/main/java/org/wso2/carbon/identity/api/server/organization/configs/common/factory/OrganizationConfigManagerOSGIServiceFactory.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    protected OrganizationConfigManager createInstance() throws Exception {
+
+        if (this.organizationConfigManager == null) {
+            OrganizationConfigManager taskOperationService = (OrganizationConfigManager) PrivilegedCarbonContext.
+                    getThreadLocalCarbonContext().getOSGiService(OrganizationConfigManager.class, null);
+            if (taskOperationService != null) {
+                this.organizationConfigManager = taskOperationService;
+            } else {
+                throw new Exception("Unable to retrieve OrganizationConfigManager service.");
+            }
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.common/src/main/java/org/wso2/carbon/identity/api/server/organization/configs/common/factory/OrganizationConfigManagerOSGIServiceFactory.java:49]: if null throw the exception,
remove else {} block and write the rest of the code

File: components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.common/src/main/java/org/wso2/carbon/identity/api/server/organization/configs/common/factory/OrganizationConfigManagerOSGIServiceFactory.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public Class<?> getObjectType() {
+
+        return Object.class;
+    }
+
+    @Override
+    protected OrganizationConfigManager createInstance() throws Exception {
+
+        if (this.organizationConfigManager == null) {
+            OrganizationConfigManager taskOperationService = (OrganizationConfigManager) PrivilegedCarbonContext.
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.common/src/main/java/org/wso2/carbon/identity/api/server/organization/configs/common/factory/OrganizationConfigManagerOSGIServiceFactory.java:43]:
```suggestion
            OrganizationConfigManager organizationConfigManagerService = (OrganizationConfigManager) PrivilegedCarbonContext.
```

File: components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.v1/src/main/resources/organization-configs.yaml [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      responses:
+        '200':
+          description: Successful response.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Config'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '403':
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.v1/src/main/resources/organization-configs.yaml:44]: Isn't that 404 expected?

File: components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.v1/src/main/resources/organization-configs.yaml [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+          example: Some Error Description
+          description: An error description.
+        traceId:
+          type: string
+          example: e0fbcfeb-3617-43c4-8dd0-7b7d38e13047
+          description: An error trace identifier.
+
+    Config:
+      type: object
+      properties:
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.v1/src/main/resources/organization-configs.yaml:131]: shouldn't this "properties" attribute marked as a required property of "Config" schema?

File: components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/configs/v1/core/OrganizationConfigsService.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    private static final Log LOG = LogFactory.getLog(OrganizationConfigsService.class);
+
+    /**
+     * Add the organization discovery configuration in the primary organization.
+     *
+     * @param config The organization discovery configuration.
+     */
+    public void addDiscoveryConfiguration(Config config) {
+
+        List<ConfigProperty> configProperties = config.getProperties().stream()
```
Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/configs/v1/core/OrganizationConfigsService.java:55]: Based on https://github.com/wso2/identity-api-server/pull/491/files#r1333410620 comment, 
check whether config.getProperties() != null check is required

Comment (AnuradhaSK) [inline components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/configs/v1/core/OrganizationConfigsService.java:55]: Check whether there is any possibility for "configProperties" being null due to 
https://github.com/wso2/identity-api-server/pull/491/files#r1333410620

File: components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.v1/src/main/resources/organization-configs.yaml (hunk 4 above)
Comment (dewniMW) [reply components/org.wso2.carbon.identity.api.server.organization.configs/org.wso2.carbon.identity.api.server.organization.configs.v1/src/main/resources/organization-configs.yaml:133]: fixed

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**